### PR TITLE
Add more consistency tests

### PR DESCRIPTION
**Pull request by ilevkivskyi**
Follow up fo #4971

### CODE REVIEW

**Review by hauntsaninja — COMMENTED** [review]
Might make sense to add a check that all stdlib modules have an entry in VERSIONS

File: tests/check_consistent.py [hunk 1]
```diff
@@ -16,7 +20,64 @@
...
 
-def main():
+def assert_stubs_only(directory):
+    """Check that given directory contains only valid stub files."""
+    top = directory.split(os.sep)[-1]
+    assert top.isidentifier(), "Bad directory name: {}".format(top)
+    for _, dirs, files in os.walk(directory):
+        for file in files:
+            name, ext = os.path.splitext(file)
+            assert name.isidentifier(), "Files must be valid modules"
```
Comment (JelleZijlstra) [inline tests/check_consistent.py:30]: Should add the name to the assertion failure to make the failure more debuggable. Same for a number of other checks below.

(Also, is there a reason not to use f-strings here?)

File: tests/check_consistent.py [hunk 2]
```diff
@@ -16,7 +20,64 @@
...
+            assert ext == ".pyi", "Only stub flies allowed. Got: {} in {}".format(file, directory)
+        for subdir in dirs:
+            assert subdir.isidentifier(), "Directories must be valid packages"
```
Comment (JukkaL) [inline tests/check_consistent.py:33]: Also include the name of the subdir in the message.

File: tests/check_consistent.py [hunk 3]
```diff
@@ -16,7 +20,64 @@
...
+            name, ext = os.path.splitext(entry)
+            assert name.isidentifier(), "Bad file name in stdlib"
+            assert ext == ".pyi", "Unexpected file in stdlib/@python2 root"
+        else:
+            assert_stubs_only(os.path.join("stdlib/@python2", entry))
+
+
+def check_stubs():
+    for distribution in os.listdir("stubs"):
+        assert not os.path.isfile(distribution), "Only directories allowed in stubs"
```
Comment (JukkaL) [inline tests/check_consistent.py:58]: Show `distribution` in the error message.

File: tests/check_consistent.py [hunk 4]
```diff
@@ -16,7 +20,64 @@
...
+        for entry in os.listdir(os.path.join("stubs", distribution)):
+            if os.path.isfile(os.path.join("stubs", distribution, entry)):
+                name, ext = os.path.splitext(entry)
+                if ext != ".pyi":
+                    assert entry in {"METADATA.toml", "README", "README.md", "README.rst"}, entry
+                else:
+                    assert name.isidentifier(), "Bad file name in stubs"
```
Comment (JukkaL) [inline tests/check_consistent.py:65]: Include name in the error message.

File: tests/check_consistent.py [hunk 5]
```diff
@@ -16,7 +20,64 @@
...
+            else:
+                if entry == "@python2":
+                    continue
+                assert_stubs_only(os.path.join("stubs", distribution, entry))
+        if os.path.isdir(os.path.join("stubs", distribution, "@python2")):
+            for entry in os.listdir(os.path.join("stubs", distribution, "@python2")):
+                if os.path.isfile(os.path.join("stubs", distribution, "@python2", entry)):
+                    name, ext = os.path.splitext(entry)
+                    assert name.isidentifier(), "Bad file name in stubs"
+                    assert ext == ".pyi", "Unexpected file in @python2 stubs"
```
Comment (JukkaL) [inline tests/check_consistent.py:75]: Include file name in the above two error messages.

File: tests/check_consistent.py [hunk 6]
```diff
@@ -35,4 +96,6 @@ def main():
 
 
 if __name__ == "__main__":
-    main()
+    check_stdlib()
+    check_stubs()
+    check_same_files()
```
Comment (JukkaL) [inline tests/check_consistent.py:101]: Should we also validate METADATA.toml files here?

**Review by JukkaL — APPROVED** [review]
Thanks for adding the tests! These will help prevent all sorts weird failure modes that would otherwise be possible, such as breaking the build pipeline.

Looks good, mostly left comments about missing details in error messages.

File: tests/check_consistent.py (hunk 6 above)
Comment (ilevkivskyi) [reply tests/check_consistent.py:101]: I think yes. I will add a test for it too.

File: tests/check_consistent.py [hunk 7]
```diff
@@ -34,5 +97,62 @@ def main():
...
+    assert not extra, f"Versions not in modules: {extra}"
+
+
+def check_metadata():
+    for distribution in os.listdir("stubs"):
+        with open(os.path.join("stubs", distribution, "METADATA.toml")) as f:
+            data = toml.loads(f.read())
+        assert "version" in data, f"Missing version for {distribution}"
+        version = data["version"]
+        msg = f"Unsupported Python version{version}"
```
Comment (JukkaL) [inline tests/check_consistent.py:137]: Nit: add space after 'version'?

**Review by JukkaL — APPROVED** [review]
Thanks for the updates! Can you create a follow-up issue about better dependency tests?

**Pull request merged**